admin/database/postgres/postgres.go

```diff
@@ -180,25 +180,31 @@ func (c *connection) FindProjectsForUser(ctx context.Context, userID string) ([]
 	return res, nil
 }
 
-func (c *connection) FindProjectsForOrganization(ctx context.Context, orgID string) ([]*database.Project, error) {
+func (c *connection) FindProjectsForOrganization(ctx context.Context, orgID string, paginationOpts *database.PaginationOptions) ([]*database.Project, error) {
 	var res []*database.Project
-	err := c.getDB(ctx).SelectContext(ctx, &res, "SELECT p.* FROM projects p WHERE p.org_id=$1 ORDER BY lower(p.name)", orgID)
+	err := c.getDB(ctx).SelectContext(ctx, &res, "SELECT p.* FROM projects p WHERE p.org_id=$1 AND lower(p.name) > $2 ORDER BY lower(p.name) LIMIT $3",
+		orgID,
+		paginationOpts.Cursor,
+		paginationOpts.PageSize)
 	if err != nil {
 		return nil, parseErr("projects", err)
 	}
 	return res, nil
 }
 
-func (c *connection) FindProjectsForOrgAndUser(ctx context.Context, orgID, userID string) ([]*database.Project, error) {
+func (c *connection) FindProjectsForOrgAndUser(ctx context.Context, orgID, userID string, paginationOpts *database.PaginationOptions) ([]*database.Project, error) {
 	var res []*database.Project
 	err := c.getDB(ctx).SelectContext(ctx, &res, `
-		SELECT p.* FROM projects p
-		WHERE p.org_id = $1 AND p.id IN (
-			SELECT upr.project_id FROM users_projects_roles upr WHERE upr.user_id = $2
+		select up.* from ( SELECT p.* FROM projects p
+			WHERE p.org_id = $1 AND p.id IN (
+				SELECT upr.project_id FROM users_projects_roles upr WHERE upr.user_id = $2
+				UNION
+				SELECT ugpr.project_id FROM usergroups_projects_roles ugpr JOIN usergroups_users uug ON ugpr.usergroup_id = uug.usergroup_id WHERE uug.user_id = $2
+			)
 			UNION
-			SELECT ugpr.project_id FROM usergroups_projects_roles ugpr JOIN usergroups_users uug ON ugpr.usergroup_id = uug.usergroup_id WHERE uug.user_id = $2
-		)
-	`, orgID, userID)
+			SELECT p.* FROM projects p WHERE p.org_id = $1 AND p.public = true
+		) up where lower(up.name) > $3 ORDER BY lower(up.name) LIMIT $4
+	`, orgID, userID, paginationOpts.Cursor, paginationOpts.PageSize)
```

Comment:
1. Wrapping `SELECT` and second union is redundant – just move to `(p.public = true OR p.id IN ...`
2. Capitalize SQL keywords